tests/std/tests/GH_001010_filesystem_error_encoding/test.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <filesystem>
+#include <string_view>
+#include <system_error>
+
+using namespace std;
+namespace fs = std::filesystem;
+
+void test_filesystem_error_with_bad_codepage_characters() {
+    fs::path problem_path{L"problematic\xD83Dtestpath"}; // path containing lone high surrogate
+
+    // Make sure the path is not encodeable in our current locale, otherwise the test tests nothing
+    bool exception_caught = false;
+    try {
+        (void) problem_path.string();
+    } catch (std::exception const&) {

```

Comment:
Nitpicks: This test is already `using namespace std;`. Also, while we previously followed a convention of "every test gets to decide whether it's east `const` or west `const`", we're probably going to change the tests to be consistently west `const` like product code (simpler to explain to new contributors), so we may as well follow that for new tests.
```suggestion
    } catch (const exception&) {
```